test/alternator/test_batch.py

```
@@ -614,6 +615,91 @@ def test_batch_get_item_partial(scylla_only, dynamodb, rest_api, test_table_sn):
+# Per-table setup must finish before the first read is launched, so a setup
+# exception cannot abandon a previously submitted read. Reproduces #31295.
+def test_batch_get_item_setup_failure_precedes_reads(scylla_only, rest_api, test_table_sn):
+    item = {'p': random_string(), 'c': 0}
+    test_table_sn.put_item(Item=item)
+    request_items = {test_table_sn.name: {'Keys': [item], 'ConsistentRead': True}}
+
+    setup_failure_injection = "alternator_batch_get_item_setup_failure"
+    read_launch_injection = "alternator_batch_get_item_read_launch"
+    with scylla_inject_error(rest_api, read_launch_injection) as read_launch:
+        with scylla_inject_error(rest_api, setup_failure_injection, one_shot=True):
+            with pytest.raises(ClientError, match="InternalServerError"):
```

Ruff (0.16.2)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)